src/librustc/front/map/mod.rs

```diff
@@ -571,6 +641,11 @@ impl<'ast> Map<'ast> {
     fn with_path_next<T, F>(&self, id: NodeId, next: LinkedPath, f: F) -> T where
         F: FnOnce(PathElems) -> T,
     {
+        // This function reveals the name of the item and hence is a
+        // kind of read. This is inefficient, since it walks ancestors
+        // and we are walking them anyhow, but whatever.
+        self.read(id);
```

Comment:
So, I've changed my mind about this one: The read should be there. If something depends on the name, it must be invalidated by a name change. The only thing different when changing the name (as opposed to changing some other part of the item) is that the change will look to the system like them item was removed instead of altered. But that does not change the fact that we need the edge to properly invalidate things depending on it.